src/vchan_cmds.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2024 EPAM Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <domain.h>
+#include <vch.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/shell/shell.h>
+
+LOG_MODULE_REGISTER(vchan_cmd);
+
+#define ECHO_SERVER_THREAD_PRIO    7
+static struct k_thread echo_thread;
+static struct k_thread *pecho_thread = NULL;
+static struct vch_handle handle;
+
+static K_THREAD_STACK_DEFINE(main_stack, 4096);
```

Comment:
Pls,

1.  add Kconfig option here to define thread stack 
2. use some more informative name for stack - smth. like `echo_thread_stack`